dpnp/dpnp_algo/dpnp_elementwise_common.py

```diff
@@ -2452,6 +2494,46 @@ def dpnp_real(x, out=None, order="K"):
     return dpnp_array._create_from_usm_ndarray(res_usm)
 
 
+_reciprocal_docstring = """
+reciprocal(x, out=None, order="K")
+
+Computes the reciprocal of each element `x_i` for input array `x`.
+
+Args:
+    x (dpnp.ndarray):
+        Input array, expected to have a real-valued floating-point data type.
+    out ({None, dpnp.ndarray}, optional):
+        Output array to populate.
+        Array must have the correct shape and the expected data type.
+    order ("C", "F", "A", "K", optional):
+        Memory layout of the newly output array, if parameter `out` is ``None``.
+        Default: "K".
+Returns:
+    dpnp.ndarray:
+        An array containing the element-wise reciprocals.
+        The returned array has a floating-point data type determined
+        by the Type Promotion Rules.
+"""
+
+reciprocal_func = _make_unary_func(
+    "reciprocal", dpt.reciprocal, _reciprocal_docstring
+)
+
+
+def dpnp_reciprocal(x, out=None, order="K"):
+    """Invokes reciprocal() from dpctl.tensor implementation for reciprocal() function."""
+
+    # dpctl.tensor only works with usm_ndarray
+    x1_usm = dpnp.get_usm_ndarray(x)
+    out_usm = None if out is None else dpnp.get_usm_ndarray(out)
+
+    res_usm = reciprocal_func(x1_usm, out=out_usm, order=order)
+    if out is None:
+        return dpnp_array._create_from_usm_ndarray(res_usm)
+    else:
+        return out
```

Comment:
@antonwolfy 
Previously, we were only using `return dpnp_array._create_from_usm_ndarray(res_usm)` here. However, that causes a problem when we have `out` keyword. In that case, the output of the function and the output coming from `out` keyword are two different copies of the same data. I have modified the code to the one here to make output of the function a view of the output coming from `out` keyword. If this is fine, we need to change all other elementwise function as well.

Reply:
Good catch! Yes, I think it would be good to properly update all other functions by separate PR.